### PR TITLE
feat: Sentry setup in energy

### PR DESCRIPTION
**Pull request by rajsam003**
This PR resolves [MOPLAT-323]

Here is the [project console ](https://sentry.io/organizations/artsynet/projects/energy/?issuesType=new&project=1382627) to check the errors

### PR Checklist

- [x] I tested my changes on **iOS** and **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

Need help with something? Have a look at our [docs], or get in touch with us.

[docs]: /docs/README.md

[MOPLAT-323]: https://artsyproduct.atlassian.net/browse/MOPLAT-323?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/system/hooks/useErrorReporting.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  Sentry.init({
+    dsn: Config.SENTRY_DSN,
+    release: energySentryReleaseName(),
+    dist: getBuildNumber(),
+    enableAutoSessionTracking: true,
+    autoSessionTracking: true,
+    enableOutOfMemoryTracking: false,
+    ...props,
+  })
+}
```
Comment (damassi) [inline src/app/system/hooks/useErrorReporting.ts:32]: We should move this setup code into `system/devTools/sentry.ts` (sentry is a dev tool) but then keep the hook defined below here in this file.

File: src/app/system/hooks/useErrorReporting.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+import { setupSentry } from "app/system/devTools/setupSentry"
+import { GlobalStore } from "app/system/store/GlobalStore"
+
+export function useErrorReporting() {
+  const environment = GlobalStore.useAppState((store) => store.config.environment.activeEnvironment)
+
+  useEffect(() => {
+    setupSentry({ environment })
+  }, [environment])
+}
```
Comment (damassi) [inline src/app/system/hooks/useErrorReporting.ts:11]: 🙏

File: android/sentry.properties [hunk 3]
```diff
@@ -0,0 +1,4 @@
+defaults.url=https://sentry.io/
+defaults.org=artsynet
+defaults.project=folio
+auth.token=12f8b79f2775433c89005aba7b3c41584365f62aaa364803a14ff5cd787a4119
```
Comment (brainbicycle) [inline android/sentry.properties:4]: is this a sensitive value 👀

File: fastlane/Fastfile [hunk 4]
```diff
@@ -188,94 +186,116 @@ lane :promote_beta_ios do
...
+  end
+
+  if dist_version.nil?
+    UI.user_error!("Sentry distribution version not specified")
+  end
+
+  source_map_path = ''
+  bundle_path = ''
+  outfile = ''
+  if platform == "ios"
```
Comment (pvinis) [inline fastlane/Fastfile:209]: this if can go, since we dont have emission on energy

File: fastlane/Fastfile [hunk 5]
```diff
@@ -188,94 +186,116 @@ lane :promote_beta_ios do
...
+    source_map_path = 'emission/Pod/Assets/Emission.js.map'
+    bundle_path = 'emission/Pod/Assets/Emission.js'
+    outfile = '~/Emission.js'
+  else
```
Comment (pvinis) [inline fastlane/Fastfile:213]: ah but this should stay. the if part goes, the else part stays and becomes the main code

File: fastlane/Fastfile [hunk 6]
```diff
@@ -188,94 +186,116 @@ lane :promote_beta_ios do
...
+      puts e.message
+      puts e.backtrace.join("\n\t")
+  end
+
+  puts "Created a release for #{project_slug}"
+
+  if platform == "ios"
+    # make individual dSYM archives available to the sentry-cli tool.
+    root = File.expand_path('..', __dir__)
+    dsym_archive = File.join(root, 'Artsy.app.dSYM.zip')
```
Comment (pvinis) [inline fastlane/Fastfile:248]: not Artsy. maybe Folio?

File: ios/sentry.properties [hunk 7]
```diff
@@ -0,0 +1,4 @@
+defaults.url=https://sentry.io/
+defaults.org=artsynet
+defaults.project=folio
+auth.token=12f8b79f2775433c89005aba7b3c41584365f62aaa364803a14ff5cd787a4119
```
[the remark this answers is not in the document]
Comment (rajsam003) [reply ios/sentry.properties]: I am gonna delete them and fetch it from S3

**Review by pvinis — APPROVED** [review]
seems good 👍. lets merge and we can QA on beta.

**Pull request merged**